crates/subspace-node/src/import_blocks_from_dsn.rs

```diff
@@ -80,9 +81,29 @@ impl ImportBlocksFromDsnCmd {
             }),
         );
 
-        import_blocks(&node, client, &mut import_queue, false)
-            .await
-            .map_err(Into::into)
+        let mut imported_blocks = 0;
+
+        // Repeat until no new blocks are imported
+        loop {
```

Comment:
Is there some way it can get stuck in the loop (as new blocks keep showing up)?

Reply:
The only way this could happen if it takes longer to retrieve one segment than to produce new segment (128 MiB of data), which while possible, also means node wouldn't be able to sync/keep up either way.

Nodes will only store last few segments worth of blocks for catch-up sync, so node must be able to pull blocks faster than they are produced for this to work at all.